autowarefoundation/autoware.universe
# feat(default_ad_api): add localization api

## Description


Provide standard API of localization score for external applications. 

## Related links



## Tests performed


1. Download sample data for rosbag replay simmulation. [Link](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/rosbag-replay-simulation/)
2. Run logging simulator.
3. Open another terminal and replay the rosbag.
4. Open another terminal and run the following command. `ros2 launch default_ad_api default_ad_api.launch.py`
5. Check the output by the command `ros2 topic echo /default_ad_api/node/api/get/localization_scores`

The result will be displayed as follows.

> values:
> - name: transform_probability
>   value: 5.909358501434326
> - name: nearest_voxel_transformation_likelihood
>   value: 3.06380295753479

## Notes for reviewers



## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.


## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.


## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.


After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/
